blobfuse/src/DataLakeBfsClient.cpp

```diff
@@ -325,24 +325,30 @@ std::vector<std::string> DataLakeBfsClient::Rename(const std::string /*sourcePat
 
 std::vector<std::string> DataLakeBfsClient::Rename(std::string sourcePath, std::string destinationPath)
 {
+    std::vector<std::string> file_paths_to_remove;
+
     errno = 0;
     m_adls_client->move_file(
             configurations.containerName,
             sourcePath.substr(1),
             configurations.containerName,
             destinationPath.substr(1));
+    
+    if(errno != 0) {
+        syslog(LOG_ERR, "Failure to rename source file %s in container.  Errno = %d.\n", sourcePath.c_str(), errno);
+    } else {
+        std::string srcMntPathString = prepend_mnt_path_string(sourcePath);
+        std::string dstMntPathString = prepend_mnt_path_string(destinationPath);
 
-    std::vector<std::string> file_paths_to_remove;
-    std::string srcMntPathString = prepend_mnt_path_string(sourcePath);
-    std::string dstMntPathString = prepend_mnt_path_string(destinationPath);
-
-    long int rename_ret = rename_cached_file(srcMntPathString.c_str(), dstMntPathString.c_str());
+        long int rename_ret = rename_cached_file(srcMntPathString.c_str(), dstMntPathString.c_str());
 
-    if(rename_ret != 0)
-    {
-        syslog(LOG_ERR, "Failure to rename source file %s in the local cache.  Errno = %d.\n", sourcePath.c_str(), errno);
+        if(rename_ret != 0) {
+            syslog(LOG_ERR, "Failure to rename source file %s in the local cache.  Errno = %d.\n", sourcePath.c_str(), errno);
+        }
+        
+        file_paths_to_remove.push_back(sourcePath);
+        errno = 0;
```

Comment:
1)
rename_cached_file should not return a non zero errno if rename_ret =0. Please fix line add errno=0 after 473 of rename_cached_file 
2) this line doesn't make sense, to simply reset the errno, if the file is in local cache and rename_cached_file could not rename there is something wrong the error has to bubble up.
3) same with file_paths_to_remove, if the local rename has failed when it is in cache what will we achieve by adding the oldpath to the list to be removed from cache .

Reply:
- Code modified to take care of point 1,2.
- For point 3 : the files are renamed in container but local rename has failed. If we keep the files in local cache directory listing will show both of them.